### ISSUE TITLE
Design decisions for Pyslurm update

### ISSUE THREAD

**Issue by supermanue** [gingergeeks/pyslurm#39]
Hi all,

This is the thread where I ask you obvious stuff about pyslurm architecture until you get bored of me. I am newbie both with pyslurm and with Cython, so probably some of the questions are kind of obvious at first. Sorry for that.

This said, 

1) I have the impression that all the methods from slurm.h are defined in slurm.pdx. Is this correct?

2) ... but not all the constants and parameters, like
```
/* Open stdout/err file mode, 0 for system default (JobFileAppend) */
#define OPEN_MODE_APPEND	1
#define OPEN_MODE_TRUNCATE	2
```
Why?

3) Then, pyslurm.pyx does not implement all methods from slurm.pdx, only a subset. Why? Is there any criteria on that? 
This applies for example to 
```
#
    # JobSteps
    #

    cdef extern int slurm_get_job_steps (time_t, uint32_t, uint32_t, job_step_info_response_msg_t **, uint16_t)
    cdef extern void slurm_free_job_step_info_response_msg (job_step_info_response_msg_t *)
    cdef extern slurm_step_layout_t *slurm_job_step_layout_get (uint32_t, uint32_t)
    cdef extern void slurm_job_step_layout_free (slurm_step_layout *)
    cdef extern int slurm_job_step_stat (uint32_t, uint32_t, char *, job_step_stat_response_msg_t **)
    cdef extern int slurm_job_step_get_pids (uint32_t, uint32_t, char *, job_step_pids_response_msg_t **)
    cdef extern void slurm_job_step_pids_free (job_step_pids_t *)
    cdef extern void slurm_job_step_pids_response_msg_free (void *)
    cdef extern void slurm_job_step_stat_free (job_step_stat_t *)
    cdef extern void slurm_job_step_stat_response_msg_free (void *)
    cdef extern int slurm_update_step (step_update_request_msg_t * step_msg)
```

4) Some of the methods from pyslurm.pyx keep the naming from the original C library. Others, however, have been renamed.  It seems that slurm.pdx keeps the naming from slurm.h, but pyslurm.pyx creates a new one. For example

```
slurm_job_cpus_allocated_on_node to __cpus_allocated_on_node
slurm_job_cpus_allocated_on_node_id to __cpus_allocated_on_node_id
```

5) The tests tend to fail. I am correcting them to make them work, but I have the impression that they are kind of old, so they are not expected to work with the lastest version of your library. Is this correct, or am I doing something wrong?

I think this is all for now.  Thanks for your help,

Manuel

**Comment by supermanue** [comment]
Hi, 

Sorry for my late answer. I was moved out of this project for a while (change on the priorities of my office) and then had some Christmas vacations.  Anyway, I hope to be working on this again form now on.

As a first step I am integrating the job submission and control functions with your previous work. After that, maybe working in a high level layer (as we discussed before) would be a good idea. 

See you!
